dao/src/main/java/org/thingsboard/server/dao/sql/event/JpaBaseEventDao.java

```diff
@@ -192,7 +192,14 @@ public PageData<? extends Event> findEventByFilter(UUID tenantId, UUID entityId,
     public void removeEvents(UUID tenantId, UUID entityId, Long startTime, Long endTime) {
         log.debug("[{}][{}] Remove events [{}-{}] ", tenantId, entityId, startTime, endTime);
         for (EventType eventType : EventType.values()) {
-            getEventRepository(eventType).removeEvents(tenantId, entityId, startTime, endTime);
+            Map<Long, Long> partitions = partitioningRepository.getPartitions(eventType.getTable(),
+                    partitionConfiguration.getPartitionSizeInMs(eventType));
+            partitions.forEach((startTs, endTs) -> {
+                if (startTs <= endTime && endTs > startTime) {
```

Comment:
Did not get the condition....
Possible combinations seems not covered
```
Sa  ... Sp1 ... Ep1 ... Ea
Sa  ... Sp1 ... Ep ...  Ep1
Sp1 ... Sa  ... Ea ...  Ep1
Sp1 ... Sa  ... Ep1 ... Ea
```


I believe all comparison should be inclusive `>= <=`

Reply:
Partition end ts is exclusive, but query start and end time are both inclusive.